src/tools/compiletest/src/main.rs

```diff
@@ -496,6 +499,14 @@ fn configure_lldb(config: &Config) -> Option<Config> {
 }
 
 pub fn test_opts(config: &Config) -> test::TestOpts {
+    match env::var("RUST_TEST_NOCAPTURE") {
+        Ok(_val) => eprintln!(
+            "WARNING: RUST_TEST_NOCAPTURE is no longer used. \
+                   Use the `--nocapture` flag instead."
+        ),
+        _ => (),
+    }
```

Comment:
`if env::var("RUST_TEST_NOCAPTURE").is_ok()` will be enough with less nested codeblock